### DISCUSSION TITLE
Is it possible to maintain newlines in .sentences.out()?

### DISCUSSION THREAD

**Discussion by Chaddeus** [winkjs/wink-nlp#114] [Help on coding] [answered]
For example, let's say the following paragraphs are read into a doc.

> The light began to dim, emphasizing the foreboding feeling that had descended over the room. All at once a strange rumble filled the room, growing louder until its intensity threatened to break the fragile silence that had taken residence within the walls. Everyone held their breath, wondering what had caused the strange sound and what it would bring them. Suddenly, a figure appeared in the center of the room, shrouded in a thick mist that seemed to grow more intense with every passing second.
> 
> The shape that emerged from the fog was a horrifying sight; it was a gruesome amalgamation of flesh and metal, its skeletal-like body covered by a layer of burned skin that gave the impression of centuries spent in the furnace. Its skin was twisted into horrific shapes, as if melted and remolded by claws of demons. Beside its face sat a skull-like mask, which helped to conceal its actual face and only gave the impression that it was smiling cruelly.

If I do `doc.sentences.out('array')` I get what I'm looking for, almost. I want to maintain that newline between paragraphs. So that if later the array were joined, the newline would remain.

I looked through the docs, including the `its` helper. But maybe I'm missing something.

Thanks in advance.

**Comment by sanjayaksaxena** [comment, marked as answer]
You are right, there is no helper to get that. You can use following code to meet your requirements:

```javascript
const winkNLP = require( 'wink-nlp' );
const model = require( 'wink-eng-lite-web-model' );
const nlp = winkNLP( model );
const its = nlp.its;

const text = 'U.S.A is my birth place.   \n\nI was born on 06.12.1924.I climbed Mt. Everest.    It was fun!';
const doc = nlp.readDoc( text );
doc.sentences().each(s => {
  const sentence = [];
  s.tokens().each( t => sentence.push( t.out(its.precedingSpaces), t.out() ));
  console.log(sentence.join(''))
});

// Outputs:
// --> "U.S.A is my birth place."
// --> "   \n\nI was born on 06.12.1924."
// --> "I climbed Mt. Everest."
// --> "    It was fun!"
```

A similar discussion took place [here](https://github.com/winkjs/wink-nlp/discussions/88). Any suggestions on what kind of a meaningful helper can be added for such needs would be appreciated.

Thanks.

**Reply by Chaddeus** [reply]
Perfect. That works great. Thank you.

In terms of the other conversation. I like having an as.helper -- that would be great. as.untrimmedText or maybe as.rawText. Any way you add it will be excellent I'm sure.

Thanks for the quick reply.
